### DISCUSSION TITLE
Converting from gemmi.cif.Block to gemmi.ReflnBLock

### DISCUSSION THREAD

**Discussion by LuisA92** [project-gemmi/gemmi#318] [Q&A]
Hello, 

I was wondering if there is a way to convert a `gemmi.cif.Block` to a `gemmi.ReflnBlock`. 

I would like to do the following: 

```python
doc = gemmi.cif.read('/.cif')
rblock = gemmi_cif.sole_block()
mtz = gemmi.CifToMtz().convert_block_to_mtz(rblock)
```

**Comment by wojdyr** [comment]
There is such a function, assuming that you mean mmCIF files, but it takes cif.Document. The second line in your example would be:

    rblock = gemmi.as_refln_blocks(doc)[0]

The reason why it takes Document not Block is that sometimes, if multiple blocks are present, the subsequent blocks in the file might be missing the space group and cell parameters. In such cases, the function copies them from the first block in the Document. (This approach might be questionable, actually.)

For small molecule files, we have function `hkl_cif_as_refln_block` that takes cif.Block.
